reflect.go

```diff
@@ -1020,6 +1016,19 @@ func ignoredByJSONSchemaTags(tags []string) bool {
 	return tags[0] == "-"
 }
 
+// toJSONNumber converts string to *json.Number.
+// It'll aso return whether the number is valid.
+func toJSONNumber(s string) (json.Number, bool) {
```

Comment:
Not sure the `bool` is necessary given that you can test for an empty number, but we can go with that.